presto-delta/src/main/java/com/facebook/presto/delta/DeltaPageSourceProvider.java

```diff
@@ -216,7 +221,21 @@ private static ConnectorPageSource createParquetPageSource(
 
         ParquetDataSource dataSource = null;
         try {
-            FSDataInputStream inputStream = hdfsEnvironment.getFileSystem(user, path, configuration).open(path);
+            ExtendedFileSystem filesystem = hdfsEnvironment.getFileSystem(user, path, configuration);
+            FileStatus fileStatus = filesystem.getFileStatus(path);
+
+            HiveFileContext hiveFileContext = new HiveFileContext(
+                    true,
+                    NO_CACHE_CONSTRAINTS,
+                    Optional.empty(),
+                    OptionalLong.of(fileSize),
+                    OptionalLong.of(start),
+                    OptionalLong.of(length),
+                    fileStatus.getModificationTime(),
+                    false);
+
+            FSDataInputStream inputStream = hdfsEnvironment.getFileSystem(user, path, configuration).openFile(path, hiveFileContext);
```

Comment:
do we need to repeat the call to `hdfsEnvironment.getFileSystem` here again?
Can't we use the same `fileSystem` from line 224.

Reply:
Good catch, yes we can reuse it. I will update